platforms/software/testing-base/src/main/java/org/gradle/api/tasks/testing/CompositeTestEventGenerator.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.testing;
+
+import org.gradle.api.Incubating;
+
+/**
+ * Generates test events, and child test event generators.
+ *
+ * @since 8.12
+ */
+@Incubating
+public interface CompositeTestEventGenerator extends TestEventGenerator {
+    /**
+     * Create a nested solitary test event generator. This can be used, for example, to add a node for each method in a tested class.
+     *
+     * <p>
+     * Since this is a solitary node, it will not have any children. To add children, use {@link #createCompositeNested(String)}.
```

Comment:
Maybe instead of "solitary", we could say "Leaf"?  

The naming is a little odd because "composite nested" sounds redundant and "solitary nested" sounds contradictory.

Reply:
Yeah, I was thinking of using "leaf" but then I wanted to be consistent and use "branch" for composite tests instead, but that conflicts with our existing terminology (`TestDescriptor#isComposite()`). It seems like we already use `ATOMIC` for `JvmTestKind` in tooling API, perhaps we should use that here too?